django_project/monitor/observation_views.py

```diff
@@ -73,16 +74,20 @@ def create_observations(request):
             try:
                 site = Sites.objects.get(gid=site_id)
             except Sites.DoesNotExist:
-                # If it doesn't exist, create a new site
+                # If it doesn't exist, create a new site with an incremented ID
+                max_site_id = Sites.objects.all().aggregate(Max('gid'))['gid__max']
+                new_site_id = max_site_id + 1 if max_site_id is not None else 1
+
                 site_name = datainput.get('siteName', '')
                 river_name = datainput.get('riverName', '')
                 description = datainput.get('siteDescription', '')
                 river_cat = datainput.get('rivercategory', '')
                 longitude = datainput.get('longitude', 0)
                 latitude = datainput.get('latitude', 0)
 
-                # Save the new site
+                # Save the new site with the incremented ID
                 site = Sites.objects.create(
+                    gid=new_site_id,
```

Comment:
@NyakudyaA yes it would but the problem is it assumes the next auto increment value is 7 from the original existing records in the db but because of the data we loaded those ids were taken so it has to find the maximum existing id then increment from there to skip the existing ids